wweval/R/ms_fig4.R

```diff
@@ -484,14 +486,78 @@ make_fig4_rel_crps_by_phase <- function(scores) {
     geom_hline(aes(yintercept = 1), linetype = "dashed") +
     xlab("Epidemic phase") +
     ylab("Relative CRPS") +
-    scale_y_continuous(trans = "log10") +
+    scale_y_continuous(trans = "log10", limits = c(0.5, 2)) +
     get_plot_theme(
       x_axis_title_size = 8,
       y_axis_title_size = 8
     ) +
-    scale_fill_brewer(palette = "Set3")
+    scale_fill_manual(values = colors$phase_colors)
+
+  return(p)
+}
+#' Make figure that plots distribution of crps scores stratified by
+#' epidemic phase
+#'
+#' @param scores A tibble of scores by location, forecast date, date and model,
+#' containing the outputs of `scoringutils::score()` on samples plus metadata
+#' transformed into a tibble.
+#'
+#' @return A ggplot object containing plots of the distribution of
+#' CRPS scores, colored by model, stratified by epidemic phase, across
+#' locations and forecast dates
+#' @export
+make_sfig_crps_by_phase <- function(scores) {
+  scores_w_fig_order <- scores |>
+    order_phases()
+
+  # Quick warning if there are NAs in epidemic phases
+  missing_phases <- scores |>
+    dplyr::filter(is.na(phase))
```

Comment:
This is supposed to reproduce the older plot that was aggregated across horizons within a location, yes? Perhaps I'm missing something, but I don't think it will. The code looks equivalent to the disaggregated main figure above.

Reply:
I accidentally copied over the computing of the relative crps but it isn't being used. The difference here is that we are making the stathalfeye plot colored by model and with the y axis value by crps, instead of relative crps

Reply:
Got it, thanks!